src/event-cache/EventCache.ts

```diff
@@ -130,8 +141,12 @@ export class EventCache {
             // Return all events.
             rumEvents = this.events;
             this.events = [];
+            this.store.clear();
         } else {
             // Dispatch the front of the array and retain the back of the array.
+            for (let i = 0; i < this.config.batchLimit; i++) {
+                this.store.evictById(this.events[i].id);
+            }
```

Comment:
issue: I agree that we need an eviction policy to limit the memory consumed by the web client. A problem with evicting records that are being dispatched is that web vitals may not have completed by the time the event is evicted.

A few other options we could explore:
1. Is there a way for us to only retain the events related to web vitals attribution?
2. Should **only** record events related to web vitals attribution?
3. Would a simple eviction policy like **first N** or **most recent N** be suitable? 

Regarding (3), maybe we can reason about which would be suitable based on the LCP algorithm.

Reply:
1. Some optimization on event **storing** is already done, but I am looking into how to reduce this to the bare minimum
2. I did not consider reducing event **recording** to only those related to web vitals. That would be a major behavior change and require a separate PR. 
3. As we discussed offline, yes I think a simpler eviction strategy is better. Currently troubleshooting if we can get away with retaining "first N rum events".

Reply:
Regarding (1), here are the optimizations
1. Only record the first navigation event, which describes the page that contains the LCP resource. We can ignore every subsequence navigation event. 
2. Record every image resource up until LCP is reported, then find the matching URL and ignore everything else. We must record everything because any of the images could be the largest contentful paint, and we have no way of knowing until LCP is reported.